src/NHibernate/Persister/Collection/IQueryableCollection.cs

```diff
@@ -91,4 +95,43 @@ public interface IQueryableCollection : IPropertyMapping, IJoinable, ICollection
 		[Obsolete("Use directly the alias parameter value instead")]
 		string GenerateTableAliasForKeyColumns(string alias);
 	}
+
+	public static class QueryableCollectionExtensions
+	{
+		/// <summary>
+		/// Gets the select fragment containing collection element, index and indentifier columns.
+		/// </summary>
+		/// <param name="queryable">The <see cref="IQueryableCollection"/> instance.</param>
+		/// <param name="alias">The table alias.</param>
+		/// <param name="columnSuffix">The column suffix.</param>
+		/// <returns>The element, index and indentifier select fragment.</returns>
+		// 6.0 TODO: Move into IQueryableCollection
+		public static SelectFragment GetSelectFragment(this IQueryableCollection queryable, string alias, string columnSuffix)
+		{
+			if (queryable is AbstractCollectionPersister collectionPersister)
+			{
+				return collectionPersister.GetSelectFragment(alias, columnSuffix);
+			}
+
+#pragma warning disable 618
+			var renderedText = queryable.SelectFragment(alias, columnSuffix);
+#pragma warning restore 618
+			var identifierAlias = queryable.GetIdentifierColumnAlias(null);
+			var indexAliases = queryable.GetIndexColumnAliases(null);
+			var columnAliases = queryable.GetKeyColumnAliases(null)
+			                             .Union(queryable.GetElementColumnAliases(null));
```

Comment:
Same.. Better to `Concat` and then once `Distinct`

Reply:
Done as suggested.